### PR TITLE
close the audiocontext in the destroy method of the webaudio backend

### PR DESCRIPTION
**Pull request by mspae**
There is a maximum amount (6) for active audioContexts, so this allows on the fly enabling/disabling without reaching that limit (if the number of currently active acs is always below 6)

https://developer.mozilla.org/en/docs/Web/API/AudioContext/close

### CODE REVIEW

File: src/webaudio.js [hunk 1]
```diff
@@ -257,6 +257,7 @@ WaveSurfer.WebAudio = {
         this.gainNode.disconnect();
         this.scriptNode.disconnect();
         this.analyser.disconnect();
+        this.ac.close();
```
Comment (katspaugh) [inline src/webaudio.js]: If you don't want to remove the `ac` parameter now, we could track when it's passed externally (vs created internally) and close or not close based on whether we created it.

Comment (mspae) [inline src/webaudio.js]: But that wouldn't fix use case #1 when an outside script is accessing WaveSurfer.WebAudio.audioContext  – Although I think that is probably a very highly exotic use case: using the audioContext, even after destroying the wavesurfer instance. – I think we could probably get away with doing what you propose without a large fallout.

Comment (mspae) [inline src/webaudio.js]: Something like this: https://github.com/katspaugh/wavesurfer.js/pull/862/commits/3c9a2bab729c75488814039a99084c1815bc9029

**Comment by thijstriemstra** [comment]
Why are the tests failing? Seems like this started when the audiocontext code was changed.

File: src/webaudio.js (hunk 1 above)
Comment (katspaugh) [inline src/webaudio.js]: Just to clarify, what I mean is that we need to remove the caching.

**Comment by katspaugh** [comment]
Yes, the tests started failing after this change of mine: 09f25635b3897641aab010ae9d3f37dd05d9959d

I'll take a look.

**Comment by mspae** [comment]
ah, I see :)

**Comment by katspaugh** [comment]
Sorry, pressed the wrong button.

File: src/webaudio.js [hunk 2]
```diff
@@ -10,15 +10,6 @@ WaveSurfer.WebAudio = {
         return !!(window.AudioContext || window.webkitAudioContext);
     },
 
-    getAudioContext: function () {
```
Comment (katspaugh) [inline src/webaudio.js]: Please don't remove the function itself, it's used here: https://github.com/katspaugh/wavesurfer.js/blob/33e01a6be465d328d0712a7836091fd6efc6d869/plugin/wavesurfer.microphone.js

**Comment by katspaugh** [comment]
Merged, thanks! I'll make a release when I fix the build.

**Comment by thijstriemstra** [comment]
Please update the changelog as well.

**Comment by arianrhodsandlot** [comment]
I'm running into some promlems with v1.3.0, so I'm now using some workarounds in my project. However it seems this pull request already solved my problem... Will a new version with these changes be released recently? Thanks for your hard works!

**Pull request merged**